cdap-gateway/src/main/java/co/cask/cdap/gateway/router/handlers/SecurityAuthenticationHttpHandler.java

```diff
@@ -384,4 +409,22 @@ public boolean find(ChannelBuffer buffer, int guessedIndex) {
       return true;
     }
   };
+
+  private static final ChannelBufferIndexFinder TWO_SEPERATOR_FINDER = new ChannelBufferIndexFinder() {
+    private byte[] lineSeperator  = "\r\n\r\n".getBytes(Charsets.UTF_8);
```

Comment:
From https://tools.ietf.org/html/rfc7230#section-2.7.3, the body is after two CRLFs, which means two `\r\n`.